thread/common/omrthreadmem.cpp

```diff
@@ -58,11 +59,11 @@ typedef struct J9ThreadMemoryHeader {
 void
 omrthread_mem_init(omrthread_library_t threadLibrary)
 {
-	memcpy(&threadLibrary->threadLibraryCategory, &threadCategoryTemplate, sizeof(OMRMemCategory));
-	memcpy(&threadLibrary->nativeStackCategory, &nativeStackCategoryTemplate, sizeof(OMRMemCategory));
+	new (&threadLibrary->threadLibraryCategory) OMRMemCategory(threadCategoryTemplate);
```

Comment:
This doing a new using the copy constructor?

Reply:
Yeah, placement new with a copy constructor